docs/source/add_new_model.md

```diff
@@ -0,0 +1,573 @@
+
+We expect that every model added to 🤗Transformers passes a couple of integration tests, meaning that the original 
+model and the reimplemented version in 🤗Transformers have to give the exact same output up to a precision of 0.001! 
+It is not enough if the model gives nearly the same output, they have to be the same. Therefore, you will certainly compare the intermediate outputs of the 🤗Transformers version multiple times against the intermediate outputs of the original implementation of brand_new_bert in which case an **efficient** debugging environment of the original repository is absolute key. Here is some advice is to make your debugging environment as efficient as possible.
+	
+- Find the best way of debugging intermediate results. Is the original repository written in PyTorch? Then you should be able to use a simple debugger like [ipdb](https://pypi.org/project/ipdb/) to print out intermediate values. Is the original repository written in Tensorflow 1? Then you might have to rely on TensorFlow print operations like 
+  https://www.tensorflow.org/api_docs/python/tf/print to output intermediate values. Is the original repository written in Jax? Then make sure that the model is **not jitted** when running the forward pass, 
+  *e.g.* check-out [this link](https://github.com/google/jax/issues/196).
+- Use the smallest pre-trained checkpoint you can find. The smaller the checkpoint, the faster your debug cycle becomes. It is not efficient if your pre-trained model is so big that your forward pass takes more than 10 seconds. In case only very large checkpoints 
+  are available, it might make more sense to create a dummy model in the new environment with randomly initialized weights 
+  and save those weights for comparison with the 🤗Transformers version of your model
+- Make sure you are using the easiest way of calling a forward pass in the original repository. Ideally, you want to find the function
+  in the original repository that **only** calls a single forward pass, *i.e.* that is often called `predict`, `evaluate`, `forward` or `__call__`. 
+  You don't want to debug a function that calls `forward` multiple times, *e.g.* to generate text, like `autoregressive_sample`, `generate`.
+- Try to separate the tokenization from the model's `forward` pass. If the original repository shows examples where you have to input a string, then try to find out where in the forward call the string input is changed to input ids and start from this point. This might mean that you have to possibly write a small script yourself or change the original code so that you can directly input the ids instead of an input string.
+- Make sure that the model in your debugging setup is **not** in training mode, which often causes the model to yield random outputs due to multiple dropout layers in the model. Make sure that the forward pass in your debugging environment is **deterministic** so that the dropout layers are not used.
```

Comment:
Or use `transformers.file_utils.set_seed` if the old and new implementations are in the same framework.